### DISCUSSION TITLE
proto2 OK,buf proto3 error.

### DISCUSSION THREAD

**Discussion by zhangjinming** [protobuf-c/protobuf-c#642] [Q&A]
syntax = "proto3";
package gamesvr;
option java_package = "com.heclouds.onenet.game";

enum MessageType {
	HEART_BEAT = 0;
	GAME_START = 1;
	GAME_END = 2;
	REQUEST_TOKEN = 3;
	RESPONSE_TOKEN = 4;
	PUSH_DATA = 5; // one way
	REQUEST_SCORE = 6;
	RESPONSE_SCORE = 7;
}

message TokenResponse {
	string url = 1;
	bytes img = 2; 
}

message Score {
	int32 score = 1;
	int32 hitNum = 2;
}

message Rank {
	int32 rank = 1;
	string name = 2;
	int32 score = 3;
	int64 time = 4;
	int32 hitNum = 5;
}

message MotionData {
	int64 time = 3;
	double alpha = 4;
	double beta = 5;
	double gamma = 6;
	double x = 7;
	double y = 8;
	double z = 9; 
}

message GameMessage {
	int32 seqNo = 1;
	MessageType messageType = 2;
	string token = 3;
	TokenResponse tokenResponse = 4;
	MotionData motionData = 5;
	Score score = 6;
	Rank rank = 7;
}

proto2 Parsing and Serialization success,buf proto3 Serialization size is error.

**Comment by pbor** [comment]
proto3 support is implemented on branch "next"
